eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelay.scala

```diff
@@ -265,21 +266,25 @@ class NodeRelay private(nodeParams: NodeParams,
         case WrappedPaymentSent(paymentSent) =>
           context.log.debug("trampoline payment fully resolved downstream")
           success(upstream, fulfilledUpstream, paymentSent)
-          stopping()
+          stopping(upstream.adds.map(_.receivedAt).minOption.getOrElse(TimestampMilli.now()), isSucess = true)
         case WrappedPaymentFailed(PaymentFailed(_, _, failures, _)) =>
           context.log.debug(s"trampoline payment failed downstream")
           if (!fulfilledUpstream) {
             rejectPayment(upstream, translateError(nodeParams, failures, upstream, nextPayload))
           }
-          stopping()
+          stopping(upstream.adds.map(_.receivedAt).minOption.getOrElse(TimestampMilli.now()), fulfilledUpstream)
       }
     }
 
   /**
    * Once the downstream payment is settled (fulfilled or failed), we reject new upstream payments while we wait for our parent to stop us.
    */
-  private def stopping(): Behavior[Command] = {
+  private def stopping(timestampBegin: TimestampMilli, isSucess: Boolean): Behavior[Command] = {
     parent ! NodeRelayer.RelayComplete(context.self, paymentHash, paymentSecret)
+    Metrics.RelayedPaymentDuration
```

Comment:
Same comment as the channel relay case, I think it would be more interesting to record this metric only in the cases where we actually send HTLCs downstream, since we want to measure how long the downstream path is taking to resolve them. That means you should be able to remove this new argument to `stopping()` if setting the metric is extracted to a separate function.

Reply:
Done. I now only measure the duration of the downstream part (including the path-finding time) but excluding the wait for the multiple parts upstream.